### PR TITLE
[MM-34340] Migrated RhsCardHeader component to TypeScript

### PR DESCRIPTION
**Pull request by djanda97**
#### Summary

Migrated RhsCardHeader component to TypeScript

#### Ticket Link

Fixes https://github.com/mattermost/mattermost-server/issues/17263
JIRA: https://mattermost.atlassian.net/browse/MM-34340

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: components/rhs_card_header/rhs_card_header.tsx [hunk 1]
```diff
@@ -10,21 +9,21 @@ import OverlayTrigger from 'components/overlay_trigger';
...
+        showFlaggedPosts: () => void;
+        showPinnedPosts: () => void;
+        closeRightHandSide: () => void;
+        toggleRhsExpanded: () => void;
     };
+};
 
-    handleBack = (e) => {
+export default class RhsCardHeader extends React.PureComponent<Props> {
+    handleBack = (e: { preventDefault: () => void }): void => {
```
Comment (djanda97) [inline components/rhs_card_header/rhs_card_header.tsx:26]: I feel like there is a better way to type `e`. What would be the preferred type?

Comment (devinbinnie) [reply components/rhs_card_header/rhs_card_header.tsx:26]: Should be of type `React.MouseEvent<HTMLAnchorElement>`

File: components/rhs_card_header/index.tsx [hunk 2]
```diff
@@ -2,7 +2,9 @@
 // See LICENSE.txt for license information.
 
 import {connect} from 'react-redux';
-import {bindActionCreators} from 'redux';
+import {AnyAction, bindActionCreators, Dispatch} from 'redux';
+
+import {GlobalState} from 'types/store/index.js';
```
Comment (devinbinnie) [inline components/rhs_card_header/index.tsx:7]: This can just be `types/store`

File: components/rhs_card_header/rhs_card_header.tsx [hunk 3]
```diff
@@ -10,21 +9,21 @@ import OverlayTrigger from 'components/overlay_trigger';
...
-        actions: PropTypes.shape({
-            showMentions: PropTypes.func,
-            showSearchResults: PropTypes.func,
-            showFlaggedPosts: PropTypes.func,
-            showPinnedPosts: PropTypes.func,
-            closeRightHandSide: PropTypes.func,
-            toggleRhsExpanded: PropTypes.func.isRequired,
-        }),
+type Props = {
+    previousRhsState: string;
```
Comment (devinbinnie) [inline components/rhs_card_header/rhs_card_header.tsx:13]: There's actually an `RhsState` type in `types/store/rhs`

**Review by devinbinnie — COMMENTED** [review]
Looks good other than a couple comments, thanks @djanda97!

**Review by devinbinnie — APPROVED** [review]
LGTM! Thanks @djanda97!

**Review by jwilander — APPROVED** [review]
Nice 👌 thanks @djanda97 !

**Pull request merged**